commandapi-core/src/main/java/dev/jorel/commandapi/CommandAPIHandler.java

```diff
@@ -161,21 +163,44 @@ Command<Source> generateCommand(Argument[] args, CommandAPIExecutor<CommandSende
 		// Generate our command from executor
 		return (cmdCtx) -> {
 			AbstractCommandSender<? extends CommandSender> sender = platform.getSenderForCommand(cmdCtx, executor.isForceNative());
+			CommandArguments commandArguments = argsToCommandArgs(cmdCtx, args);
+			AbstractExecutionInfo<CommandSender, AbstractCommandSender<? extends CommandSender>> executionInfo = new AbstractExecutionInfo<>() {
+				@Override
+				public CommandSender sender() {
+					return sender.getSource();
+				}
+
+				@Override
+				public CommandArguments args() {
+					return commandArguments;
+				}
+			};
 			if (converted) {
-				Object[] argObjs = argsToObjectArr(cmdCtx, args);
 				int resultValue = 0;
 
 				// Return a String[] of arguments for converted commands
 				String[] argsAndCmd = cmdCtx.getRange().get(cmdCtx.getInput()).split(" ");
 				String[] result = new String[argsAndCmd.length - 1];
+				AbstractExecutionInfo<CommandSender, AbstractCommandSender<? extends CommandSender>> convertedExecutionInfo = new AbstractExecutionInfo<CommandSender, AbstractCommandSender<? extends CommandSender>>() {
+					@Override
+					public CommandSender sender() {
+						return sender.getSource();
+					}
+
+					@Override
+					public CommandArguments args() {
+						return new CommandArguments(result, new LinkedHashMap<>());
```

Comment:
Wait... This wouldn't work, would it?

Reply:
For more context, please take a look at the file. I don't know how to specify multiple lines (if that is possible at all)

Reply:
Yes, this should work. Converted commands don't (and can't) access arguments by node name, they only access them via index

Reply:
Hmm... Okay, thanks. I thought it wouldn't as I pass the new String[] into that method which hasn't been filled with content yet...